solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -2503,7 +2503,8 @@ public static int persistConfigResourceToZooKeeper(ZkSolrResourceLoader zkLoader
   public static void touchConfDir(ZkSolrResourceLoader zkLoader) {
     SolrZkClient zkClient = zkLoader.getZkController().getZkClient();
     try {
-      zkClient.setData(zkLoader.getConfigSetZkPath(), new byte[]{0}, true);
+      byte[] currentData = zkClient.getData(zkLoader.getConfigSetZkPath(), null, null, true);
+      zkClient.setData(zkLoader.getConfigSetZkPath(), currentData, true);
```

Comment:
Probably a race condition here?

Reply:
Do you have a suggestion here?   I'm going through and manually testing this to confirm the problem....    However, if you had a code fix, I'd love to see it pushed up ;-)

Reply:
I don't have an IDE in front of me, but there's a flag to setData I think that lets us fail if the node already exists to avoid overwriting the current data.

Reply:
I addressed this in https://github.com/apache/solr/pull/831 - updated for main